api/src/main/java/marquez/db/migrations/V44_1__BackfillAirflowParentRuns.java

```diff
@@ -0,0 +1,193 @@
+ * many Airflow tasks report the ParentRunFacet as <code>
+ *   "parentRun": {
+ *         "job": {
+ *           "name": "simple_shortrunning_dag.tasks__1_of_3",
+ *           "namespace": "abcdefg"
+ *         },
+ *         "run": {
+ *           "runId": "scheduled__2022-03-14T01:40:10+00:00"
+ *         }
+ *       }
+ * </code> The RunId can't be mapped to an existing Marquez run and the job name is actually a
+ * concatenation of the DAG name and the task (which is the name assigned to the job for the task
+ * itself).
+ *
+ * <p>This migration aims to create a real parent run, with a distinct run id, and parent job for
+ * each such run by constructing a deterministic run id from the DAG name and its runId. From each
+ * run, a job is created with the name field set to the DAG name.
+ */
+@Slf4j
+public class V44_1__BackfillAirflowParentRuns implements JavaMigration {
+
+  /**
+   * Return a numeric version that is greater than 44 (so it executes after that one) but before 45
+   */
+  public static final MigrationVersion MIGRATION_VERSION = MigrationVersion.fromVersion("44.1");
+
+  private static final String FIND_AIRFLOW_PARENT_RUNS_SQL =
+      """
+      SELECT DISTINCT(run_uuid) AS run_uuid,
+                     e.parent_run_id,
+                     e.parent_job_name,
+                     e.parent_job_namespace
+      FROM runs r
+      LEFT JOIN LATERAL (
+          SELECT run_uuid,
+                 event->'run'->'facets'->'parent'->'run'->>'runId' AS parent_run_id,
+                 event->'run'->'facets'->'parent'->'job'->>'name' AS parent_job_name,
+                 event->'run'->'facets'->'parent'->'job'->>'namespace' AS parent_job_namespace
+          FROM lineage_events le
+          WHERE le.run_uuid=r.uuid
+          AND event->'run'->'facets'->'parent'->'run'->>'runId' IS NOT NULL
+          AND event->'run'->'facets'->'airflow_version' IS NOT NULL
+          ) e ON e.run_uuid=r.uuid
+      WHERE e.parent_run_id IS NOT NULL
+""";
+  public static final String INSERT_PARENT_JOB_QUERY =
+      """
+      INSERT INTO jobs (uuid, type, created_at, updated_at, namespace_uuid, name, description,
+      namespace_name, current_location)
+      SELECT :uuid, type, created_at, updated_at, namespace_uuid, :name, description, namespace_name,
+      current_location
+      FROM jobs
+      WHERE namespace_name=:namespace AND name=:jobName
+      ON CONFLICT(name, namespace_uuid) WHERE parent_job_uuid IS NULL
+      DO UPDATE SET updated_at=now()
+      RETURNING uuid
+      """;
+  public static final String INSERT_PARENT_RUN_QUERY =
+      """
+      INSERT INTO runs (uuid, created_at, updated_at, current_run_state, external_id, namespace_name, job_name, job_uuid, location, transitioned_at, started_at, ended_at)
+      SELECT :parentRunUuid, created_at, updated_at, current_run_state, :externalRunid, :namespace, :jobName, :parentJobUuid, location, transitioned_at, started_at, ended_at
```

Comment:
_Typo_: `externalRunid` -> `externalRunId` (though looks like it still works, so maybe more of a naming convention thing)